app/src/hooks/useOngoingTransferCleaner.tsx

```diff
@@ -0,0 +1,47 @@
+import { NotificationSeverity } from '@/models/notification'
+import { CompletedTransfer, StoredTransfer, TxStatus } from '@/models/transfer'
+import { getExplorerLink, startedTooLongAgo } from '@/utils/transfer'
+import { captureException } from '@sentry/nextjs'
+import { useEffect } from 'react'
+import useCompletedTransfers from './useCompletedTransfers'
+import useNotification from './useNotification'
+import useOngoingTransfers from './useOngoingTransfers'
+
+const useOngoingTransfersCleaner = (ongoingTransfers: StoredTransfer[]) => {
+  const { remove } = useOngoingTransfers()
+  const { addCompletedTransfer } = useCompletedTransfers()
+  const { addNotification } = useNotification()
+
+  useEffect(() => {
+    ongoingTransfers.forEach(ongoing => {
+      if (startedTooLongAgo(ongoing)) {
+        const explorerLink = getExplorerLink(ongoing)
+        remove(ongoing.id)
+        addCompletedTransfer({
+          id: ongoing.id,
+          result: TxStatus.Undefined,
+          token: ongoing.token,
+          sourceChain: ongoing.sourceChain,
+          destChain: ongoing.destChain,
+          amount: ongoing.amount,
+          tokenUSDValue: ongoing.tokenUSDValue ?? 0,
+          fees: ongoing.fees,
+          sender: ongoing.sender,
+          recipient: ongoing.recipient,
+          date: ongoing.date,
+          ...(explorerLink && { explorerLink }),
+        } satisfies CompletedTransfer)
+
+        addNotification({
+          message: 'Transfer verification failed.',
```

Comment:
The language here is not quite there. There is no "verification" taking place here, it's just tracking. And personally I prefer to not end these small and single lines with a `.`

```suggestion
          message: 'Transfer tracking failed',
```